### DISCUSSION TITLE
[2024-08-05] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#10747] [Weekly Demo Day]
* Worked mostly on #10293 

[Screencast from 2024-08-05 16-10-31.webm](https://github.com/user-attachments/assets/7d569d5b-4389-4a11-b523-9f396c99e9c1)

* In the meantime, fixes some issues important for the release:
  * #10654
  * #10619 (after electron bump)
  * #10721

**Comment by vitvakatu** [comment]
A bunch of non-visual fixes and bug investigations:

- https://github.com/enso-org/enso/pull/10671 – Regression of top-right menu button.
- https://github.com/enso-org/enso/pull/10673 – Fixed live editing of images in doc panel.
- https://github.com/enso-org/enso/pull/10693 – Critical issue in Code Editor.
- https://github.com/enso-org/enso/issues/10678 – Investigated, proved to be the issue with the Engine.
- https://github.com/enso-org/enso/issues/10527 – Ongoing investigation, suspected critical issue in AST editing.

**Comment by Frizi** [comment]
This iteration I was mostly focusing on ydoc-server build scripts and migration to bazel:
- Finished refactoring of node widgets. The node icon is no longer special, it is always drawn as a widget - part of https://github.com/enso-org/enso/issues/10484
- https://github.com/enso-org/enso/pull/10735 - separated parts of gui project into separate workspace-level projects. Now ydoc-server can be a standalone nodejs process with its own build pipeline.
- https://github.com/enso-org/enso/pull/10508 - Working on migrating remaining parts of GUI build pipeline to bazel. Currently stuck on issues with building rust on windows.

**Comment by kazcw** [comment]
#10473 (missing syntax errors for unexpected usages of `=`) is a deep issue. I have a fix in a PR that will be ready today. On the way there, I fixed:
- #5444
- #6152
- #10137 
- #6741 
- #8633 
- #5452 
- #5331
- Fixed `sbt`/`run` incompatibilities around parser dependencies, that caused redundant builds and "CI: Clean build needed" cases.
- Set up basic support for fuzz testing the parser.
- Improved Rust parsing performance by 33%; now 20 MB/s on my bench machine

**Comment by hubertp** [comment]
Relevant backend work:
- Disabling JVM Ydoc #10627 
- Record button for a single node not working
- Swallow stderr/stdout output from project-manager https://github.com/enso-org/enso/pull/10690
- Revert and reapply idmap changes to show widgets on startup #10626 & #10700 
- Working on emoji support
